From c00fb7222cf4015e23b1b76804846ec43aeffc1b Mon Sep 17 00:00:00 2001
From: Matheus Castanho <msc@linux.ibm.com>
Date: Thu, 30 Jan 2020 10:08:47 -0300
Subject: [PATCH] Skip machine_name fix for powerpc* on fixincludes step

Some system headers can be broken by the machine_name fix performed
by GCC during the fixincludes step. According to the comment in
fixincludes/fixinc.h:130 :

   On some platforms, machine_name doesn't work properly and
   breaks some of the header files.  Since everything works
   properly without it, just wipe the macro list to
   disable the fix.

So we can just skip it to avoid trouble.
---
 fixincludes/fixinc.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fixincludes/fixinc.in b/fixincludes/fixinc.in
index 15cbaa23544..c0791454b9c 100755
--- a/fixincludes/fixinc.in
+++ b/fixincludes/fixinc.in
@@ -136,6 +136,9 @@ fi
 #  disable the fix.
 
 case "${target_canonical}" in
+    powerpc*-*-linux-*)
+	test -f ${MACRO_LIST} &&  echo > ${MACRO_LIST}
+        ;;
     *-*-vxworks*)
 	test -f ${MACRO_LIST} &&  echo > ${MACRO_LIST}
         ;;
-- 
2.21.1

